dll/cpl/usrmgr/users.c

```diff
@@ -650,6 +651,51 @@ UpdateUserProperties(HWND hwndDlg)
     NetApiBufferFree(pUserInfo);
 }
 
+void OnToggleRequireLogon(HWND hwndDlg)
+{
+    BOOL bIsChecked;
+    WCHAR szAutoAdminLogonValue[2]; 
+    HKEY hKey;
+    LONG lResult;
+    TCHAR szErrorMsg[256];
+
+    bIsChecked = IsDlgButtonChecked(hwndDlg, IDC_USERS_STARTUP_REQUIRE);
+
+    wcscpy(szAutoAdminLogonValue, (bIsChecked == BST_CHECKED) ? L"0" : L"1");
+
+    lResult = RegCreateKeyExW(HKEY_LOCAL_MACHINE,
+                              L"SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion\\Winlogon",
+                              0,
+                              NULL,
+                              REG_OPTION_NON_VOLATILE,
+                              KEY_SET_VALUE,
+                              NULL,
+                              &hKey,
+                              NULL);
+
+    if (lResult == ERROR_SUCCESS)
+    {
+        lResult = RegSetValueExW(hKey,
+                                 L"AutoAdminLogon",
+                                 0,
+                                 REG_SZ, 
+                                 (const BYTE*)szAutoAdminLogonValue,
+                                 (wcslen(szAutoAdminLogonValue) + 1) * sizeof(WCHAR));
+
+        if (lResult != ERROR_SUCCESS)
+        {
+            wsprintf(szErrorMsg, L"Failed to set AutoAdminLogon registry value. Error code: %ld", lResult);
+            MessageBoxW(hwndDlg, szErrorMsg, L"Registry Error", MB_OK | MB_ICONERROR);
```

Comment:
You can also make use of `DPRINT` macros to allow these errors to be visible during debugging sessions. Formatting codes reference can be found [here](https://reactos.org/wiki/DPRINT_codes) and how that works [here](https://reactos.org/wiki/Debugging#ReactOS_Style). Whether the use of DPRINT is recommended in this case, I'll let other reviewers decide on this.